src/MainWindow.vala

```diff
@@ -214,24 +219,35 @@ public class AppCenter.MainWindow : Gtk.ApplicationWindow {
         homepage.page_loaded.connect (() => homepage_loaded ());
     }
 
-    public override bool delete_event (Gdk.EventAny event) {
-        int window_width;
-        int window_height;
-        int window_x;
-        int window_y;
-        get_size (out window_width, out window_height);
-        unowned Settings saved_state = Settings.get_default ();
-        saved_state.window_width = window_width;
-        saved_state.window_height = window_height;
-        get_position (out window_x, out window_y);
-        saved_state.window_x = window_x;
-        saved_state.window_y = window_y;
-        if (is_maximized) {
-            saved_state.window_state = Settings.WindowState.MAXIMIZED;
-        } else {
-            saved_state.window_state = Settings.WindowState.NORMAL;
+    public override bool configure_event (Gdk.EventConfigure event) {
+        if (configure_id != 0) {
+            GLib.Source.remove (configure_id);
         }
 
+        configure_id = Timeout.add (100, () => {
+            configure_id = 0;
+
+            if (is_maximized) {
+                settings.set_boolean ("window-maximized", true);
+            } else {
+                settings.set_boolean ("window-maximized", false);
+
+                Gdk.Rectangle rect;
+                get_allocation (out rect);
+                settings.set ("window-size", "(ii)", rect.width, rect.height);
+
+                int root_x, root_y;
+                get_position (out root_x, out root_y);
+                settings.set ("window-position", "(ii)", root_x, root_y);
+            }
+
+            return false;
+        });
+
+        return base.configure_event (event);
```

Comment:
Can you try with `return false` instead? The documentation explicitly stands that:
> true to stop other handlers from being invoked for the event. false to propagate the event further.

Reply:
This does really fucky things when you resize the window. Decorations get clipped and other weird things

Reply:
Alright, thanks for trying :)